OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbyUtils.cs

```diff
@@ -272,21 +272,27 @@ public static void SetupEditableNameWidget(Widget parent, Session.Slot s, Sessio
 			name.IsDisabled = () => orderManager.LocalClient.IsReady;
 
 			name.Text = c.Name;
-			name.OnLoseFocus = () =>
+			name.OnEscKey = () =>
+			{
+				name.Text = c.Name;
+				Ui.Root.Get<TextFieldWidget>("CHAT_TEXTFIELD").TakeKeyboardFocus();
```

Comment:
This is a generic utility function, so shouldn't make any assumptions about the UI state.  You could instead pass in a `defaultFocus` widget, and then give that focus if it is non-null.